### PR TITLE
refactor: add routing on teams page

### PR DESCRIPTION
**Pull request by thesanjeevsharma**
fixes: #1095 

This has a slightly different pattern than rest of the pages because team names had spaced words unlike repository names. Also some teams have special chars like - `.` and `-`. 

I am open for suggestions, if you have any.

### CODE REVIEW

**Review by jamessspanggg — APPROVED** [review]
LGTM

File: src/containers/Teams/index.tsx [hunk 1]
```diff
@@ -1,40 +1,53 @@
...
-  {pathname: TeamName.newUserOperations, title: 'New User Operations'},
-  {pathname: TeamName.payments, title: 'Payments'},
-  {pathname: TeamName.projectProposals, title: 'Project Proposals'},
-  {pathname: TeamName.qa, title: 'QA'},
-  {pathname: TeamName.redditModerators, title: 'Reddit Moderators'},
-  {pathname: TeamName.research, title: 'Research'},
-  {pathname: TeamName.security, title: 'Security'},
-  {pathname: TeamName.youtube, title: 'YouTube'},
+export const TEAMS: NavOption[] = [
+  {pathname: 'all', title: TeamName.all},
```
Comment (jamessspanggg) [inline src/containers/Teams/index.tsx:13]: Is it possible to refactor the team path names as constants? Just realised that the `ProfileInfo` page might need to link to the specific team's page using the path names

Comment (thesanjeevsharma) [reply src/containers/Teams/index.tsx:13]: Yeah, sure. I'll put an object in constants/teams.ts and then reference those here. Cool?

Comment (jamessspanggg) [reply src/containers/Teams/index.tsx:13]: Yep sounds good

**Review by lapstjup — APPROVED** [review]

**Review by jamessspanggg — APPROVED** [review]

**Review by Hristijan95 — APPROVED** [review]

**Pull request merged**